packages/shared/src/components/Feed.tsx

```diff
@@ -198,23 +204,28 @@ export default function Feed<T>({
 
   const showScrollOnboardingVersion =
     sidebarRendered &&
+    feedName === MainFeedPage.Popular &&
+    !isLoading &&
     alerts?.filter &&
+    !user?.id &&
     Object.values(ScrollOnboardingVersion).includes(scrollOnboardingVersion);
-  const fetchPageFn = showScrollOnboardingVersion
-    ? () => setShowFeedFilters(true)
-    : fetchPage;
+  const shouldScrollBlock =
+    sidebarRendered &&
+    [ScrollOnboardingVersion.V1, ScrollOnboardingVersion.V2].includes(
+      scrollOnboardingVersion,
+    );
 
   const infiniteScrollRef = useFeedInfiniteScroll({
-    fetchPage: fetchPageFn,
-    canFetchMore,
+    fetchPage,
+    canFetchMore: canFetchMore && !shouldScrollBlock,
   });
 
   const onInitializeOnboardingClick = () => {
     trackEvent({
       event_name: AnalyticsEvent.ClickScrollBlock,
       target_id: scrollOnboardingVersion,
     });
-    onInitializeOnboarding();
+    onInitializeOnboarding(undefined, true);
```

Comment:
I actually think we might want to redirect the user to `my-feed` if they onboarded via this way.
There is an example of that already, but maybe double check with product.

Reply:
Tested it and It already does that, not sure from where but I can check in detail.

Reply:
checked with Seb, and this already does that so nothing is needed.

Reply:
Hmm yeah the first param is the onFinish right?
I think in other implementations we do something with it.
Just to ensure that's how it works

Reply:
yeah and its undefined by default, something other redirects though.. I can find what exactly if you want?